File: src/main/java/org/example/kalkulationsprogramm/domain/SystemSetting.java
package org.example.kalkulationsprogramm.domain;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

/**
 * Schlüssel-Wert Konfiguration, die zur Laufzeit im Frontend geändert werden kann.
 * Werte überschreiben die application.properties Defaults.
 */
@Entity
@Table(name = "system_setting")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class SystemSetting {

    @Id
    @Column(name = "setting_key", length = 128)
    private String key;

    @Column(name = "setting_value", columnDefinition = "TEXT")
    private String value;

    @Column(length = 255)
    private String beschreibung;
}

File: src/main/kotlin/org/example/kalkulationsprogramm/domain/SystemSetting.kt
package org.example.kalkulationsprogramm.domain

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.Id
import jakarta.persistence.Table

@Entity
@Table(name = "system_setting")
open class SystemSetting {
    @Id
    @Column(name = "setting_key", length = 128)
    open var key: String? = null

    @Column(name = "setting_value", columnDefinition = "TEXT")
    open var value: String? = null

    @Column(length = 255)
    open var beschreibung: String? = null

    constructor()

    constructor(key: String?, value: String?, beschreibung: String?) {
        this.key = key
        this.value = value
        this.beschreibung = beschreibung
    }
}
